pyuvdata/uvdata/uvh5.py

```diff
@@ -189,259 +194,689 @@ def _get_compression(compression):
-        self.Npols = int(header["Npols"][()])
-        self.Ntimes = int(header["Ntimes"][()])
-        self.Nblts = int(header["Nblts"][()])
-        self.Nspws = int(header["Nspws"][()])
+            vis_units = "uncalib"
+        return vis_units
+
+    def to_uvdata(self, check_lsts: bool = False) -> UVData:
+        """Convert the file to a UVData object.
+
+        The object will be metadata-only.
+        """
+        uvd = UVH5()
+        uvd.read_uvh5(self, read_data=False, run_check_acceptability=check_lsts)
+        return uvd
+
+
+class UVH5(UVData):
+    """
+    A class for UVH5 file objects.
+
+    This class defines an HDF5-specific subclass of UVData for reading and
+    writing UVH5 files. This class should not be interacted with directly,
+    instead use the read_uvh5 and write_uvh5 methods on the UVData class.
+    """
+
+    def _read_header_with_fast_meta(
+        self,
+        filename: str | Path | FastUVH5Meta,
+        run_check_acceptability: bool = True,
+        blt_order: tuple[str] | None | Literal["determine"] = None,
+        blts_are_rectangular: bool | None = None,
+        time_axis_faster_than_bls: bool | None = None,
+        background_lsts: bool = True,
+    ):
+        if not isinstance(filename, FastUVH5Meta):
+            obj = FastUVH5Meta(
+                filename,
+                blt_order=blt_order,
+                blts_are_rectangular=blts_are_rectangular,
+                time_axis_faster_than_bls=time_axis_faster_than_bls,
+            )
+        else:
+            obj = filename
+
+        # First, get the things relevant for setting LSTs, so that can be run in the
+        # background if desired.
+        self.time_array = obj.time_array
+        self.telescope_location_lat_lon_alt_degrees = (
+            obj.telescope_location_lat_lon_alt_degrees
+        )
+
+        if "lst_array" in obj.header:
+            self.lst_array = obj.header["lst_array"][:]
+            proc = None
+        else:
+            proc = self.set_lsts_from_time_array(background=background_lsts)
+
+        # Required parameters
+        for attr in [
+            "instrument",
+            "telescope_name",
+            "history",
+            "vis_units",
+            "Nfreqs",
+            "Npols",
+            "Nspws",
+            "Ntimes",
+            "Nblts",
+            "Nbls",
+            "Nants_data",
+            "Nants_telescope",
+            "antenna_names",
+            "antenna_numbers",
+            "antenna_positions",
+            "ant_1_array",
+            "ant_2_array",
+            "phase_center_id_array",
+            "Nbls",
+            "baseline_array",
+            "integration_time",
+            "freq_array",
+            "spw_array",
+            "channel_width",
+            "polarization_array",
+            "uvw_array",
+            "channel_width",
+            "phase_center_catalog",
+            # "blts_are_rectangular",
+            # "time_axis_faster_than_bls",
```

Comment:
should these commented lines be removed?

Reply:
should they actually appear below in the optional parameters?

Reply:
No, they always exist, so they would always be set. I just removed them from being set in this PR as it changes a few tests. They can be set later by the user on the UVData object if desired. If you think we should set them by default, I can also do that.

Reply:
I'm confused. Why will they always exist? They wouldn't exist in older files.

Reply:
So in this bit of code, we go through this list of attributes, and for each one that exists on the `FastUVH5Meta` object, it sets it on the `UVData` object. The attributes in the `FastUVH5Meta` are dynamically-defined by the `__getattr__` method for the most part, which just goes and reads them from the file itself. However, these two attributes (`blts_are_rectangular` and `time_axis_faster_than_bls`) are cached properties on the `FastUVH5Meta` object -- they're always defined, because if they're not defined, they're calculated from the other properties. It *probably* makes sense to calculate them and set them on the UVData object, since it can make other calculations faster, but I didn't want to do this by default in this PR, since it does take a little more time to init, and also it means I have to change some tests (eg. reading a file, then writing the file and reading the written one results in a *different* final object because these new attributes are set).

Reply:
Ok, I think I follow. I've run into this kind of thing before where you can't really check if something is defined because the machinery calculates it if you ask for it. But looking at this method, I don't see where it is set on the UVH5 object if it is known from the file or has already been determined. Are we just not setting it at all even if it's in the file?

Reply:
@bhazelton You make a good point -- I wasn't setting it, even if it existed in the file itself. I've fixed that now. It will be set if and only if it exists in the file header. Otherwise, the user can set them manually on the UVData object (or, if using the `FastUVH5Meta` object directly, can access them dynamically).

Reply:
sounds good, thanks!